index.js

```diff
@@ -91,16 +91,32 @@ module.exports.iterator = (emitter, event, options) => {
 	let hasPendingError = false;
 	const nextQueue = [];
 	const valueQueue = [];
+	let eventCount = 0;
+
+	let limit = -1;
+	if (typeof options.limit === 'number') {
+		limit = options.limit;
+	}
```

Comment:
The number should be strictly type-checked, see https://github.com/sindresorhus/p-event/commit/b902979ac0a7e3ec8f4266033c4cb4560711cc1b#diff-168726dbe96b3ce427e7fedce31bb0bcR29